crates/shim-sgx/src/main.rs

```diff
@@ -275,14 +276,9 @@ global_asm!(
         CSSA_0_STK_TCS_SZ = const CSSA_0_STACK_SIZE + Page::SIZE,
 );
 
-unsafe extern "C" fn main(
-    block: &mut [usize; BLOCK_SIZE / core::mem::size_of::<usize>()],
-    ssas: &mut [StateSaveArea; 3],
-    cssa: usize,
-    tcb: &mut MaybeUninit<Tcb>,
-) -> i32 {
-    // Enable exceptions:
-    ssas[cssa].extra[0] = 1;
+fn validate_block_ptr(ptr: *mut u8) -> &'static mut [usize; BLOCK_SIZE / size_of::<usize>()] {
+    is_aligned_non_null::<[usize; BLOCK_SIZE / size_of::<usize>()]>(ptr as usize).unwrap();
+    let block = unsafe { &mut *(ptr as *mut [usize; BLOCK_SIZE / size_of::<usize>()]) };
```

Comment:
Safety comment. Is the assertion on the previous line enough to guarantee that this is safe? Can we move the construction of the block below the `if (block_start ...)` check below? Instead of `block.as_ptr() as usize` below can we just use the original `ptr` instead for that check?

Reply:
good point!

Reply:
fixed